src/tools.cc

```diff
@@ -195,40 +195,28 @@ squid_getrusage(struct rusage *r)
 #elif defined(PSAPI_VERSION)
     // Windows has an alternative method if there is no POSIX getrusage defined.
     if (WIN32_OS_version >= _WIN_OS_WINNT) {
-        /* On Windows NT and later call PSAPI.DLL for process Memory */
-        /* information -- Guido Serassio                       */
-        HANDLE hProcess;
-        PROCESS_MEMORY_COUNTERS pmc;
-        hProcess = OpenProcess(PROCESS_QUERY_INFORMATION |
-                               PROCESS_VM_READ,
-                               FALSE, GetCurrentProcessId());
-        {
-            /* Microsoft CRT doesn't have getrusage function,  */
-            /* so we get process CPU time information from PSAPI.DLL. */
-            FILETIME ftCreate, ftExit, ftKernel, ftUser;
-            if (GetProcessTimes(hProcess, &ftCreate, &ftExit, &ftKernel, &ftUser)) {
-                int64_t *ptUser = (int64_t *)&ftUser;
-                int64_t tUser64 = *ptUser / 10;
-                int64_t *ptKernel = (int64_t *)&ftKernel;
-                int64_t tKernel64 = *ptKernel / 10;
-                r->ru_utime.tv_sec =(long)(tUser64 / 1000000);
-                r->ru_stime.tv_sec =(long)(tKernel64 / 1000000);
-                r->ru_utime.tv_usec =(long)(tUser64 % 1000000);
-                r->ru_stime.tv_usec =(long)(tKernel64 % 1000000);
-            } else {
-                CloseHandle( hProcess );
-                return;
+        // On Windows NT and later call PSAPI.DLL for process Memory information
+        auto hProcess = OpenProcess(PROCESS_QUERY_INFORMATION|PROCESS_VM_READ, FALSE, GetCurrentProcessId());
+        // Microsoft CRT doesn't have getrusage function,
+        // so we get process CPU time information from PSAPI.DLL.
+        FILETIME ftCreate, ftExit, ftKernel, ftUser;
+        if (GetProcessTimes(hProcess, &ftCreate, &ftExit, &ftKernel, &ftUser)) {
+            int64_t *ptUser = (int64_t *)&ftUser;
```

Comment:
why not using auto here?

Reply:
Yes, but start with a `const auto` attempt and downgrade to `auto` if you have to.